cpp/src/parquet/column_reader.cc

```diff
@@ -1603,7 +1603,12 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     // another record start or exhausting the ColumnChunk
     int64_t level = levels_position_;
     if (at_record_start_) {
-      ARROW_DCHECK_EQ(0, rep_levels[levels_position_]);
+      if (rep_levels[levels_position_] != 0) {
```

Comment:
This looks ok to me but I forget that this will only called when `max_rep_level > 0`?

Reply:
Hmm yeah I guess there's the possibility there could be non-zero repetition levels when the maximum is supposed to be 0. I'm not sure it's worth reading the repetition levels just to check for that though?

Reply:
Let me go through this tonight. I think I can try to check this when first deducing the `at_record_start_`, but I need to go through this part of the code

Reply:
Oh let's just check it here